fixtures/datagen/fakerize.py

```diff
@@ -419,28 +521,182 @@ def augment_vasps(fake_names=FAKE_VASPS):
+    record["modified"] = end
+    record["audit_log"] = make_cert_log("COMPLETED", start, end)
+    return record
 
-if __name__ == "__main__":
 
+def make_initialized(cert, idx, template="fixtures/datagen/templates/cert_req.json"):
+    """
+    Make a cert req in the INITIALIZED state
+    """
+    with open(template, "r") as f:
+        record = json.load(f)
+
+    record["id"] = idx
+    record["vasp"] = str(uuid.uuid1())
+    common_name = make_common_name(cert)
+    record["common_name"] = common_name
+    record["status"] = "INITIALIZED"
+    start = make_dates(count=1)[0]
+    record["created"] = start
+    record["modified"] = start
+    record["audit_log"] = make_cert_log("INITIALIZED", start, start)
+    return record
+
+
+def make_cr_errored(cert, idx, template="fixtures/datagen/templates/cert_req.json"):
+    """
+    Make a cert req in the CR_ERRORED state
+    """
+    with open(template, "r") as f:
+        record = json.load(f)
+
+    record["id"] = idx
+    record["vasp"] = str(uuid.uuid1())
+    common_name = make_common_name(cert)
+    record["common_name"] = common_name
+    record["status"] = "CR_ERRORED"
+    batch = str(random.randint(100000, 999999))
+    record["batch_id"] = batch
+    network = random.choice(NETWORKS)
+    record[
+        "batch_name"
+    ] = f"{network} certificate request for {common_name} (id: {idx})"
+    record["batch_status"] = "FAILED"
+    record["order_number"] = batch
+    start, end = make_dates(count=2)
+    record["created"] = start
+    record["modified"] = start
+    record["audit_log"] = make_cert_log("CR_ERRORED", start, end)
+    return record
+
+
+def make_cr_rejected(cert, idx, template="fixtures/datagen/templates/cert_req.json"):
+    """
+    Make a cert req in the CR_REJECTED state
+    """
+    with open(template, "r") as f:
+        record = json.load(f)
+
+    record["id"] = idx
+    record["vasp"] = str(uuid.uuid1())
+    common_name = make_common_name(cert)
+    record["common_name"] = common_name
+    record["status"] = "CR_REJECTED"
+    batch = str(random.randint(100000, 999999))
+    record["batch_id"] = batch
+    network = random.choice(NETWORKS)
+    record[
+        "batch_name"
+    ] = f"{network} certificate request for {common_name} (id: {idx})"
+    record["batch_status"] = "REJECTED"
+    record["order_number"] = batch
+    start, end = make_dates(count=2)
+    record["created"] = start
+    record["modified"] = start
+    record["reject_reason"] = lorem.sentence()
+    record["audit_log"] = make_cert_log("CR_REJECTED", start, end)
+    return record
+
+
+def make_cert_log(state, start, end):
+    """
+    Return a list of dictionaries representing a synthetic but plausible audit log
+    """
+    logs = []
+
+    states = [state]
+    current_state = state
+    prior_state = None
+    while current_state in CERT_STATE_CHANGES:
+        prior_state = CERT_STATE_CHANGES[current_state]["previous_state"]
+        if prior_state == "INITIALIZED":
```

Comment:
Maybe I'm not following the logic correctly but doesn't this stop before the `INITIALIZED` state? Since the previous state for `READY_TO_SUBMIT` is also `INITIALIZED`.

Reply:
Good catch; fixed!